### ISSUE TITLE
bun init FileNotFound

### ISSUE THREAD

**Issue by ImLunaHey** [oven-sh/bun#1898]
### What version of Bun is running?

0.5.1

### What platform is your computer?

Darwin 20.6.0 x86_64 i386

### What steps can reproduce the bug?

Provide `bun init` an entry point in a non-existent directory.

### What is the expected behavior?

The directory is created.

### What do you see instead?

```
error: FileNotFound
bun could not find a file, and the code that produces this error is missing a better error.
```

**Comment by gaurishhs** [comment]
can't reproduce. It will create a file in the current directory

**Comment by ImLunaHey** [comment]
Huh? 

```
➜  bun-issue bun init
bun init helps you get started with a minimal project and tries to guess sensible defaults. Press ^C anytime to quit

package name (bun-issue):
entry point (index.ts): src/index.ts

Done! A package.json file was saved in the current directory.

error: FileNotFound
bun could not find a file, and the code that produces this error is missing a better error.

----- bun meta -----
Bun v0.5.1 (42e9f6e0) macOS x64 20.6.0
Unknown:
Elapsed: 3414ms | User: 3ms | Sys: 25ms
RSS: 3.97MB | Peak: 3.97MB | Commit: 67.11MB | Faults: 530
----- bun meta -----
```

**Comment by ImLunaHey** [comment]
Even after updating.

```
➜  bun-issue bun init
bun init helps you get started with a minimal project and tries to guess sensible defaults. Press ^C anytime to quit

package name (bun-issue):
entry point (src/index.ts): src/index.ts

Done! A package.json file was saved in the current directory.

error: FileNotFound
bun could not find a file, and the code that produces this error is missing a better error.

----- bun meta -----
Bun v0.5.5 (242dcea2) macOS x64 20.6.0
Unknown:
Elapsed: 3065ms | User: 2ms | Sys: 4ms
RSS: 4.07MB | Peak: 4.07MB | Commit: 33.55MB | Faults: 0
----- bun meta -----
```

**Comment by gaurishhs** [comment]
You have to create the `src` directory first

**Comment by ImLunaHey** [comment]
Yes... that's the bug.

**Comment by raikasdev** [comment]
I would like to bump this as I believe that the init command should create the directory.
I think it is good practice to keep your code in a seperate directory (for example src), and it would be just faster if bun simply would create that directory if it is specified in bun init.

**Comment by ImLunaHey** [comment]
Confirmed this is still an issue.

```bash
➜  paas git:(main) ✗ bun init   
bun init helps you get started with a minimal project and tries to guess sensible defaults. Press ^C anytime to quit

package name (paas): 
entry point (src/index.ts): 

Done! A package.json file was saved in the current directory.

error: FileNotFound
bun could not find a file, and the code that produces this error is missing a better error.

----- bun meta -----
Bun v0.7.1 (53cc4df1) macOS x64 21.6.0
Unknown: 
Elapsed: 2186ms | User: 4ms | Sys: 5ms
RSS: 4.26MB | Peak: 4.26MB | Commit: 33.55MB | Faults: 0
----- bun meta -----
```

**Comment by vjpr** [comment]
Maybe related to https://github.com/oven-sh/bun/issues/3863

Anything relying on `fs.open(..., {O_CREAT)` is broken...

**Comment by Electroid** [comment]
This has been fixed sometime before Bun v1.0. 

```
❯ bun init blahblah
bun init helps you get started with a minimal project and tries to guess sensible defaults. Press ^C anytime to quit

package name (code): 
entry point (index.ts): src/index.ts

Done! A package.json file was saved in the current directory.
 + src/index.ts
 + .gitignore
 + tsconfig.json (for editor auto-complete)
 + README.md

To get started, run:
  bun run src/index.ts
```

If you still run into this bug, please feel free to re-open this issue.
